### PR TITLE
NFC: enable `InferSendableFromCaptures` on almost all modules

### PR DESCRIPTION
**Pull request by MaxDesiatov**
`SourceControl` is excluded for now as it triggers concurrency errors when this feature is enabled.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

File: Package.swift [hunk 1]
```diff
@@ -845,3 +845,19 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
...
+for target in package.targets where target.type == .regular && target.name != "SourceControl" {
+    let sendableFromCaptures = SwiftSetting.enableUpcomingFeature("InferSendableFromCaptures")
+    let updatedSwiftSettings: [SwiftSetting]
+
+    if var swiftSettings = target.swiftSettings {
+        swiftSettings.append(sendableFromCaptures)
+        updatedSwiftSettings = swiftSettings
+    } else {
+        updatedSwiftSettings = [sendableFromCaptures]
+
```
Comment (MaxDesiatov) [inline Package.swift:859]:
```suggestion
```

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by xedin** [comment]
@MaxDesiatov might be a good place to drop all of the `@Sendable` annotations added on methods/properties/functions as well.

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by hi2gage** [comment]
If you replace the main return inside of `public func areIgnored(_ paths: [Basics.AbsolutePath]) throws -> [Bool]` with:

```swift 
return stringPaths.map { pathString in
    let lines = output.split(whereSeparator: { $0.isNewline })
    let processedLines = lines.map {
        let string = String($0).replacingOccurrences(of: "\\\\", with: "\\")
        if string.utf8.first == UInt8(ascii: "\"") {
            return String(string.dropFirst(1).dropLast(1))
        }
        return string
    }
     return processedLines.contains(pathString)
}
```
 Then `SourceControl` should be ready to go. I can always open another PR if that's better.

**Comment by hi2gage** [comment]
> If you replace the main return inside of `public func areIgnored(_ paths: [Basics.AbsolutePath]) throws -> [Bool]` with:
> 
> ```swift
> return stringPaths.map { pathString in
>     let lines = output.split(whereSeparator: { $0.isNewline })
>     let processedLines = lines.map {
>         let string = String($0).replacingOccurrences(of: "\\\\", with: "\\")
>         if string.utf8.first == UInt8(ascii: "\"") {
>             return String(string.dropFirst(1).dropLast(1))
>         }
>         return string
>     }
>      return processedLines.contains(pathString)
> }
> ```
> 
> Then `SourceControl` should be ready to go. I can always open another PR if that's better.

On second thought, the usage of that function was removed [here](https://github.com/swiftlang/swift-package-manager/pull/5748/files#diff-965019ff07b70555eadf67422f518a8b11623e8cc81fd50c97cf2ca59c80690eL299). That probably needs to be completely removed.

**Comment by xedin** [comment]
I talked to @MaxDesiatov  offline about this and it sounded like we are not getting much benefit by enabling this, so maybe we should close this PR?

**Pull request closed without merging**